internal/sdkv2provider/resource_cloudflare_workers_domain_test.go

```diff
@@ -0,0 +1,127 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+
+const (
+	scriptContent = `addEventListener('fetch', event => {event.respondWith(new Response('test'))});`
+)
+func TestAccCloudflareWorkerDomain_Attach(t *testing.T) {
+	var domain cloudflare.WorkersDomain
+	rnd := generateRandomResourceName()
+	hostname := rnd + ".com"
+	name := "cloudflare_worker_domain." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckAccount(t)
+		},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckCloudflareWorkerDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkerDomainAttach(rnd, accountID, hostname),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkerDomainExists(name, &domain),
+					resource.TestCheckResourceAttr(name, "hostname", hostname),
+					resource.TestCheckResourceAttr(name, "service", rnd),
+				),
+			},
+		},
+	})
+}
+
+func getDomainFromApi(accountID, domainID string) (cloudflare.WorkersDomain, error) {
+	if accountID == "" {
+		return cloudflare.WorkersDomain{}, fmt.Errorf("accountID is required to get a domain")
+	}
+	if domainID == "" {
+		return cloudflare.WorkersDomain{}, fmt.Errorf("domainID is required to get a domain")
+	}
+
+	client := testAccProvider.Meta().(*cloudflare.API)
+	domain, err := client.GetWorkersDomain(context.Background(), cloudflare.AccountIdentifier(accountID), domainID)
+	if err != nil {
+		fmt.Print(err.Error())
+		return cloudflare.WorkersDomain{}, err
+	}
+
+	return domain, nil
+}
+
+func testAccCheckCloudflareWorkerDomainExists(resourceName string, domain *cloudflare.WorkersDomain) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Worker Domain ID is set")
+		}
+
+		accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+		domainID := rs.Primary.ID
+		foundDomain, err := getDomainFromApi(accountID, domainID)
+		if err != nil {
+			return err
+		}
+
+		if foundDomain.ID != domainID {
+			return fmt.Errorf("worker domain with id %s not found", domainID)
+		}
+
+		*domain = foundDomain
+		return nil
+	}
+}
+
+func testAccCheckCloudflareWorkerDomainAttach(rnd, accountID string, hostname string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_zone" "%[1]s-zone" {
```

Comment:
why are we creating a new zone here? we have a zone already that is setup and entitled correctly for Workers so we can use that (`CLOUDFLARE_ZONE_ID` and `CLOUDFLARE_ACCOUNT_ID` respectively)